L1Trigger/TrackFindingTMTT/src/StubCluster.cc

```diff
@@ -0,0 +1,125 @@
+#include "L1Trigger/TrackFindingTMTT/interface/StubCluster.h"
+#include "L1Trigger/TrackFindingTMTT/interface/Stub.h"
+#include "L1Trigger/TrackFindingTMTT/interface/TP.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+namespace TMTT {
+
+  StubCluster::StubCluster(std::vector<const Stub *> stubs, double SectorPhi, int lres_dr) : layerKF_(999) {
+    r_ = 0;
+    phi_ = 0;
+    z_ = 0;
+    eta_ = 0;
+    sigmaX_ = 0;
+    sigmaZ_ = 0;
+    layerId_ = 0;
+    endcapRing_ = 0;
+    barrel_ = false;
+    tiltedBarrel_ = false;
+    psModule_ = false;
+    nstubs_ = stubs.size();
+    stubs_ = stubs;
+    dphi_dl_ = 0;
+    dphi_dr_ = 0;
+    dphi_ = 0;
+    deltai_ = 0;
+    alpha_ = 0;
+    moduleTilt_ = 0.;
+
+    layerIdReduced_ = stubs.at(0)->layerIdReduced();
+    layerId_ = stubs.at(0)->layerId();
+    endcapRing_ = stubs.at(0)->endcapRing();
+    barrel_ = stubs.at(0)->barrel();
+    tiltedBarrel_ = stubs.at(0)->tiltedBarrel();
+    psModule_ = stubs.at(0)->psModule();
+    moduleTilt_ += stubs.at(0)->moduleTilt();
+
+    if (nstubs_ == 0)
+      return;
+
+    double sum_r2(0), sum_z2(0), sum_phi2(0), sum_sigmaZ2(0), sum_dphi2_dr(0), sum_dphi2_dl(0);
+    int lsign(0);
+
+    for (auto stub : stubs) {
+      r_ += stub->r();
+      z_ += stub->z();
+      eta_ += stub->eta();
+      phi_ += reco::deltaPhi(stub->phi(), SectorPhi);
+      std::set<const TP *> assocTPs = stub->assocTPs();
+      if (assocTPs.size()) {
```

Comment:
```suggestion
      if (not assocTPs.empty()) {
```
anyway the check is somewhat redundant.

Reply:
Class StubCluster has been deleted